src/codeu/chat/client/core/Context.java

```diff
@@ -22,15 +22,23 @@
 import codeu.chat.common.User;
 import codeu.chat.util.Uuid;
 import codeu.chat.util.connections.ConnectionSource;
+import codeu.chat.common.ServerInfo;
+import codeu.chat.client.core.View;
 
 public final class Context {
 
   private final BasicView view;
+  // We need to access the information of the view, but since 
+  // the method that returns this is private in the child class 
+  // BasicView, we're going to create a variable ViewInterface,
+  // which has a public version of this function - Priyanka Agarwal
+  private final View viewInterface;
   private final Controller controller;
 
   public Context(ConnectionSource source) {
     this.view = new View(source);
     this.controller = new Controller(source);
+    this.viewInterface = new View(source);
```

Comment:
I'm still a little bit confused about why you need two instances of `View` here. I think I understand why you did it: because the `getInfo()` function is in the `View` class, not the `BasicView` interface, so you needed a reference to a `View` variable?

But note that the pre-existing `view` variable is already a `View` instance, so it does contain the `getInfo()` class. You just need to reference it using the subclass. The easiest ways to do that would be to just change the `view` variable to a `View` reference (change line 30 to `private final View view;`), or you could just cast it before using it (that way is a little bit messier though).

Another option would be to just add the `getInfo()` function to the `BasicView` interface. I'm not sure how feasible that is though- depends on how many other classes implement the `BasicView` interface.

Does that make sense?